students/cfarris/lessons/lesson09/assignment/mdb_db.py

```diff
@@ -0,0 +1,217 @@
+#!/usr/bin/env Python3
+
+"""
+Lesson09 part 2 Context Manager. 
+MongoDB is already in a context manager, but I did a little editing
+and put more things within the context manager.
+
+This set up is only for development use. Not production
+Uses Pymongo to access MongoDB database.
+Imports 3 files, directs them to different collections
+Retrieves information necessary for HP Norton collection
+"""
+import time
+import csv
+from pymongo import MongoClient
+
+# pylint: disable-msg=redefined-outer-name
+# pylint: disable-msg=invalid-name
+# pylint: disable-msg=too-many-locals
+
+
+class MongoDBConnection():
+    """Connect to MongoDB"""
+
+    def __init__(self, host='127.0.0.1', port=27017):
+
+        self.host = host
+        self.port = port
+        self.connection = None
+
+    def __enter__(self):
+        self.connection = MongoClient(self.host, self.port)
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.connection.close()
+        print("connection to MongoDB successfully closed...")
```

Comment:
You could add `db = mongo.connection.norton_products` to the context manager, since you always do that when you open the connection.